### DISCUSSION TITLE
starting a new trace for background job using auto java-instrumentation

### DISCUSSION THREAD

**Discussion by thiyagu06** [open-telemetry/opentelemetry-java-instrumentation#4382] [Q&A]
I have an application which polls for message in SQS and do some processing. The processing includes calling external service, DB etc.. Once I receive the message from SQS, I will submit those message to executor service to the processing. 

I used `Context.current().wrap(Executors.newFixedThreadPool(10))` this method to start tracing my message processing. But I don't see the tracing information is set in the logging MDC context? Any idea to achieve that? I'm using auto instrumentation to do the tracing.

```
Class SQSPoller(String queueName, ExecutorService executorService, MessageProcessor Processor) {
    
    public void() {
      while(true) {
        val messages = awsClient.receiveMessage();
        for(Message message: messages) {
          processor.submit({
            processor.process(message)
          })
        }
      }
      
    }
}

interface Processor {
  public void process(Message message)
}

Class DefaultProcessor implements Processor {
    public void process(Message message) {
      System.out.println("message processed")
    }
}

class Main {
  ExcetuorService executorService = Context.current().wrap(Executors.newFixedThreadPool(10))
  SQSPoller poller = new SQSPoller("message-process",executorService, new DefaultProcessor())
}
```

**Comment by trask** [comment]
hi @thiyagu06!

this:

```
Context.current().wrap(Executors.newFixedThreadPool(10))
```

creates an executor service that always runs the tasks in whatever `Context.current()` was at the time you called `.wrap()`.

and since you are calling that at the beginning of `main`, most likely it's the root context, which could explain why you're not getting propagation.

ideally you should not need to use `Context.wrap()` when you are using the javaagent.

if you are finding a case with SQS where the javaagent isn't propagating context automatically, it would be great if you can open an issue with a repro that we can use to troubleshoot it

**Reply by nic-littlepay** [reply]
> if you are finding a case with SQS where the javaagent isn't propagating context automatically, it would be great if you can open an issue with a repro that we can use to troubleshoot it

I realise this is an ancient thread but I thought it might be useful to connect this up for those following, as I came across this discussion while investigating an issue with the javaagent not propagating context automatically, except this issue is in the case of Lambda -> SQS -> Lambda.

We have the following issue open for this and it's being investigated  https://github.com/open-telemetry/opentelemetry-lambda/issues/1787
